web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java

```diff
@@ -161,7 +161,10 @@ public ResponseWithCosts doPost(@NotNull IsochroneRequest request) {
         }
         Collection<Coordinate> sites = new ArrayList<>();
         Collection<SegmentWithCost> segments = new ArrayList<>();
-        shortestPathTree.search(fromNodes, label -> {
+        // When searching from several nodes we need to use the distance as weight in the Dijkstra,
+        // so that we are guaranteed to get a lower bound on the actual distance.
+        boolean useDistanceAsWeight = request.getRegions().size() > 1;
+        shortestPathTree.search(useDistanceAsWeight, fromNodes, label -> {
```

Comment:
I'd prefer if this was a boolean option stored in the request, instead of implied like this. But it's a small thing.

Reply:
You're right, that's better!